db/view/view.cc

```diff
@@ -1070,6 +1071,7 @@ query::clustering_row_ranges calculate_affected_clustering_ranges(const schema&
             }
             for (auto&& r : v.view->view_info()->partition_slice().default_row_ranges()) {
                 view_row_ranges.push_back(r.transform(std::mem_fn(&clustering_key_prefix::view)));
+                co_await make_ready_future<>(); // yield if needed
```

Comment:
Interesting idiom, which of course will need to be document in a restructured tutorial.md. I wonder if we should have a function like maybe_yield() or something which returns make_ready_future<>() today but makes it clearer what's going on (no need for comment).

Reply:
By the way, @avikivity. how efficient is this idiom "co_await make_ready_future"? In the old futures code, checking if we need to yield is just checking a boolean. How much more expensive is this co_await, or perhaps it's optimized by the compiler to the same thing?

Reply:
It's the same thing. The compiler may introduce extra slowness from copying transient variables to the coroutine object and then back, if it doesn't consider the __builtin_expect hint.